src/ipa_pc/mod.rs

```diff
@@ -1388,7 +1481,7 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                     }
 
                     let commitment = cur_comm.commitment();
-                    combined_comm += &commitment.comm.mul(*coeff);
+                    combined_comm += &commitment.comm[0].mul(*coeff);
```

Comment:
As above: I am afraid this is not the right way to use segmented commitments in a LC. 
We need to combine all the segments (using the powers of the poly's query point).

Reply:
Done